src/measure_theory/probability_theory.lean

```diff
@@ -0,0 +1,442 @@
+  p (s₁ ∪ s₂) = p s₁ + p s₂ := measure_theory.measure_union hd h₁ h₂
+
+lemma prob_union_nnreal {s : Type u} [probability_space s] (s₁ s₂ : set s)
+(hd : disjoint s₁ s₂) (h₁ : is_measurable s₁) (h₂ : is_measurable s₂) (p : probability_measure s) :
+  prob p (s₁ ∪ s₂) = prob p s₁ + prob p s₂ :=
+begin
+  rw ← ennreal.coe_eq_coe, simp, apply prob_union s₁ s₂ hd h₁ h₂,
+end
+
+lemma prob_diff_inter {s : Type u} [probability_space s](p : probability_measure s)
+(a b : set s) (h₁ : is_measurable a) (h₂ : is_measurable b) :
+p(b ∩ -a) + p(b ∩ a) = p(b) :=
+begin
+  have h :p(b) = p(b ∩ univ),
+  by rewrite inter_univ b,
+  rewrite [h,← compl_union_self a,set.inter_distrib_left,prob_union],
+  have g : (b ∩ -a) ∩ (b ∩ a) = ∅, by rw [inter_left_comm,set.inter_assoc, compl_inter_self,inter_empty,inter_empty],
+  apply disjoint_iff.2 g,
+  {
+    rewrite ← diff_eq,
+    apply is_measurable.diff h₂ h₁,
+  },
+  apply is_measurable.inter h₂ h₁,
+end
+
+lemma prob_diff_inter_nnreal {s : Type u} [probability_space s](p : probability_measure s)
+(a b : set s) (h₁ : is_measurable a) (h₂ : is_measurable b) :
+prob p (b ∩ -a) + prob p (b ∩ a) = prob p (b) :=
+begin
+  rw ← ennreal.coe_eq_coe,
+  simp, rw add_comm,
+  apply prob_diff_inter p a b h₁ h₂,
+end
+
+lemma prob_union_inter
+(a b : set s) (g₁ : is_measurable a) (g₂ : is_measurable b) :
+p(a ∪ b) + p(a ∩ b) = p(a) + p(b) :=
+begin
+  have h₁ : a ∪ b = a ∪ (b ∩ -a),by
+    rewrite [set.union_distrib_left, union_compl_self a,inter_univ],
+  have h₂ : is_measurable(b ∩ -a),
+  {
+    rewrite ← diff_eq,
+    apply is_measurable.diff g₂ g₁,
+  },
+  rewrite [h₁,prob_union a _ _ g₁ h₂],
+  have h₃: a ∩ b = b ∩ a, by exact set.inter_comm a b,
+  rewrite [h₃,← prob_diff_inter p a b g₁ g₂],
+  simp,
+  have h₄ : a ∩ (b ∩ -a) = ∅, by tidy,
+  apply disjoint_iff.2 h₄,
+end
+
+lemma prob_union_inter_nnreal
+(a b : set s) (g₁ : is_measurable a) (g₂ : is_measurable b) :
+prob p (a ∪ b) + prob p (a ∩ b) = prob p (a) + prob p (b) :=
+begin
+  rw ← ennreal.coe_eq_coe,
+  simp only [coe_prob, ennreal.coe_add],
+  apply prob_union_inter p a b g₁ g₂,
+end
+
+/-- The Bonnferroni inequality. --/
+lemma prob_add_le_inter_add_one
+(a b : set s) (h_1 : is_measurable a) (h_2 : is_measurable b) :
+p(a) + p(b) ≤ p(a ∩ b) + 1 :=
+begin
+  rewrite [← prob_union_inter p a b h_1 h_2, ← add_comm],
+  apply add_le_add_left' _,
+  apply prob_le_1,
+end
+
+/-- The Bonnferroni inequality for a nnreal probability measure. --/
+lemma prob_add_le_inter_add_one_nnreal
+(a b : set s) (h_1 : is_measurable a) (h_2 : is_measurable b) :
+prob p a + prob p b ≤ prob p (a ∩ b) + 1:=
+begin
+  rw ← ennreal.coe_le_coe, simp only [coe_prob, ennreal.coe_add, ennreal.coe_one],
+  apply prob_add_le_inter_add_one p a b h_1 h_2,
+end
+
+def are_disj_ctble' (f : ℕ → set s) := pairwise (disjoint on λ (i : ℕ), f i)
```

Comment:
What are these primed versions of? I could not find the original versions. Why are these notions defined here?

Shouldn't we have all the theorems like `measure_bUnion` that are defined in `measure_space.lean`, and shouldn't they look exactly the same here, except that the results are in `real`?